### ISSUE TITLE
Lint/UnusedBlockArgument triggered incorrectly when using binding

### ISSUE THREAD

**Issue by JeanMertz** [bbatsov/rubocop#1637]
This one triggers `Lint/UnusedBlockArgument` incorrectly:

```ruby
tables.each do |table_name, table|
  puts ERB.new(template).result(binding)
end
```

**Comment by jonas054** [comment]
So it seems that if there is a call to `Kernel#binding` inside a block, all bets are off as far as this cop is concerned. Any variable could be accessed somewhere. Right?

Some other `Lint` cops will have to be looked at as well.

**Comment by JeanMertz** [comment]
Indeed, I believe that as soon as a `binding` call is found that is within the scope of the variables, this cop (or any other relevant to this issue) should not trigger anymore.

**Comment by bbatsov** [comment]
Sounds reasonable.
